src/algorithms/models/vertex.ts

```
@@ -67,6 +67,18 @@ class Vertex {
     });
   }
 
+  /**
+   * Get cost to neighbor (rather inefficient because neighbors is an array)
+   */
+  costTo(vertex: Vertex): EdgeCost {
```

I wouldn't call this inefficient, as I can't think of a better way how to find an element in a Set - Set has no direct accessor. In this case `vertex.neighbors` should have been a Map rather than Set for various reasons... But I don't expect vertices having many neighbors, so it's fine.

Though there is one issue, you define the output type to be an object EdgeCost, but what if you don't find the neighbor? Is it a null, or undefined? This could cause some problems in the calling parent and you should at least throw an exception (e.g. `"Vertex ${id1} has no neighbor ${id2}!"`.